kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -43,6 +43,13 @@ public static ColumnarBatch parseJsonStats(
             .parseJson(statsVector, statsSchema, scanFileBatch.getSelectionVector());
     }
 
+    /**
+     * Prunes the given schema to only include the referenced columns.
+     */
+    public static StructType pruneStatsSchema(StructType schema, Set<Column> referencedCols) {
```

Comment:
prune schema says `Prunes the given schema according to the provided set of top-level columns to keep`.

so do these `referencedCols` need to be top-level?

Reply:
Clarified

Reply:
Unfortunately, now I'm even more confused.

- `pruneSchema` says: `Prunes the given schema according to the provided set of top-level columns to keep.`
- `pruneStatsSchema` says that the `referencesCols` are the `leaf columns`

so the first asks for top-level columns, and the latter gives it leaf columns?

Reply:
`pruneStatsSchema` takes in `schema` and a set of leaf columns (for nested columns these should be the full column path)

`pruneSchema` recurses --> so the input `schema` here is nested under `parentPath`, but the `columnsToKeep` are not
- By "top-level leaf columns" I mean that the column paths are relative to the root schema and not to the current level of recursion

Reply:
Okay, I understand now (I think). I think we need to clarify this in the comments, then.

1.
```
pruneStatsSchema: Prunes the given schema to include only the referenced leaf columns. If a leaf column is nested, then it must be referenced using the full column path, e.g. `C_0.C_1.C_Leaf`
```

2. I would also rename `referencedCols` to `referencedLeafCols`

3. 
```
pruneSchema: Prunes the given schema according to include only the referenced leaf columns. These leaf columns (they can be nested, of course) are relative to the root schema, not to the current level of recursion. The `leafColumnsToKeep` doesn't change with each recursive call.

e.g.: <give an example, showing 2 layers of recursion>
```

4. if columnsToKeep doesn't change, why not mark it as final and indicate this?

5. rename to `leafColumnsToKeep`

Reply:
6. For `pruneSchema`, Seems like `parentPath` is related to `schema`, yet they are not next to each other in the param list? Let's put related parameters together. I would do: `final leafColumnsToKeep, schema, parentPath`

Reply:
Updated; I think marking `leafColumnsToKeep` as final doesn't really mean much toward it being unchanged between recursive calls (and I think you could technically modify the set anyways). Mentioned it in the docs.